server/service/mdm.go

```diff
@@ -368,10 +368,14 @@ func (r getMDMEULAMetadataResponse) Error() error { return r.Err }
 
 func getMDMEULAMetadataEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (fleet.Errorer, error) {
 	eula, err := svc.MDMGetEULAMetadata(ctx)
-	if err != nil {
+	if err != nil && !fleet.IsNotFound(err) {
 		return getMDMEULAMetadataResponse{Err: err}, nil
 	}
 
+	if eula == nil {
+		return nil, newNotFoundError()
```

Comment:
So we will still return a 404 but it won't be logged because this doesn't go through the same error-handling logic as the `fleet.Errorer` return value, right? I don't remember offhand the mechanism when we return the error via the second value, we don't really use it.

If so I think it should have a comment explaining why we do it this way. Also, it should probably have a test (it should be possible to send a mocked logger and ensure it doesn't get called to log an error, and ensure that the response is still a 404).

Reply:
I think the approach is fine with the comment explanation, we can see in the future if we have more "not found" cases where we don't want an error to be logged, then we can revisit and maybe use the `IsClientError` approach.

For the test, you can take a look at `integration_logger_test.go`, it's an integration test suite that makes a buffer available for the logs, it may be possible to add your test there: https://github.com/fleetdm/fleet/blob/26f2f13c250c66b571218be06939fd87b52eea87/server/service/integration_logger_test.go#L32-L37

Reply:
I don't know offhand, but I would also think this would be fine.